ethcore/src/blockchain/blockchain.rs

```diff
@@ -979,6 +995,34 @@ impl BlockChain {
 		}
 	}
 
+	/// Mark a block to be considered finalized. Panic if the hash does not exist.
+	pub fn mark_finalized(&self, batch: &mut DBTransaction, block_hash: H256) {
+		let mut block_details = self.block_details(&block_hash).unwrap_or_else(|| panic!("Invalid block hash: {:?}", block_hash));
```

Comment:
Maybe better to return `Result` instead of panicking?

Reply:
agree, or at least prove it won't happen

Reply:
The rest of `BlockChain` struct is also using panic for pub fns when a block hash is not found. I think for our case is that we're going to use `mark_finalized` here for results returned from Engine, and it will only call this function for already known block hashes. Otherwise, it indicates a logic error in our code. So I think it would be fine to use panic given we indicated the panic condition in the documentation.